docs/bitcoin-stake-expansion.md

```diff
@@ -0,0 +1,654 @@
+
+> **⚠️ Critical Warning**: Make sure that you are retrieving the BTC Staking
+> parameters from a trusted node and verify their authenticity using
+> additional sources. Using incorrect parameters will cause expansion
+> validation to fail or create transactions incompatible with the current
+> Babylon state.
+
+> **⚡ Parameter Selection for Expansion**
+>
+> Unlike post-staking registration which uses parameters based on Bitcoin
+> inclusion height, expansions must use parameters corresponding to the
+> **current Babylon Bitcoin light client tip** at expansion submission time.
+> This pre-commitment ensures parameter consistency even if the expansion is
+> later included in a Bitcoin block with different active parameters. The new
+> BTC stake expansion must respect the current Babylon staking parameters rules.
+
+### 3.4 Detailed Transaction Construction
+
+The Bitcoin expansion transaction and related transactions must follow the
+specific structures required by the btcstaking module. These transactions
+include:
+* **Bitcoin Expansion Transaction**: The main transaction that spends the
+  original stake and additional funding to create the expanded stake
+* **Slashing Transaction**: A pre-signed transaction consenting to slashing
+  in case of double-signing for the expanded stake
+* **Unbonding Transaction**: The on-demand unbonding transaction used to unlock
+  the expanded stake before the originally committed timelock expires
+* **Unbonding Slashing Transaction**: A pre-signed transaction consenting to
+  slashing during the unbonding process in case of double-signing
+
+#### Creating the Bitcoin Expansion Transaction
+
+The expansion transaction must follow a strict two-input structure validated 
+by the btcstaking module:
+
+**Required Transaction Structure:**
+- **Input 0**: Previous staking transaction output (the original active
+  delegation)
+  * Must reference the exact UTXO from the original staking transaction
+  * Requires covenant signatures for spending (obtained from Babylon node)
+  * Must match the `previous_staking_tx_hash` provided in the expansion message
+- **Input 1**: Additional funding UTXO (controlled by staker)
+  * Must be controlled by the same Bitcoin key used for staking
+  * Provides funds for staking value increase and transaction fees
+  * Must be from a confirmed Bitcoin transaction (the `funding_tx`)
+- **Output 0**: New expanded staking output
+  * Must follow the [staking script format](./staking-script.md) exactly
+  * Uses the expanded finality provider list (superset of original)
+  * Contains the new total staking value (≥ original amount)
+  * Must use current Babylon staking parameters
+
+#### Critical: Signing the Expansion Transaction
+
+**The expansion transaction signing process is different from regular Bitcoin
+transactions** because Input 0 (the original staking output) requires covenant
+signatures to be spent through the unbonding path.
+
```

Comment:
suggestion: Here it might be a good place to add the happy path mermaid sequential diagram from the RFC